packages/data-provider/src/utils/submissionUtils.ts

```diff
@@ -230,6 +232,23 @@ const utils = (dependencies: BaseDependencies) => {
 			);
 		},
 
+		/**
+		 * Merge two `Record<string, T[]>` objects into a single `Record<string, T[]>` object.
+		 * For each key in the records, the corresponding arrays from both records are concatenated.
+		 * @param record1 The first `Record<string, T[]>` object. If `undefined`, it is treated as an empty record.
+		 * @param record2 The second `Record<string, T[]>` object. If `undefined`, it is treated as an empty record.
+		 * @returns
+		 */
+		mergeRecords: <T>(
+			record1: Record<string, T[]> | undefined,
+			record2: Record<string, T[] | undefined>,
+		): Record<string, T[]> => {
+			return Object.keys({ ...record1, ...record2 }).reduce<Record<string, T[]>>((acc, key) => {
+				acc[key] = [...(record1?.[key] || []), ...(record2?.[key] || [])];
```

Comment:
would the items in the merged array need to be sorted somehow?

e.g. current: `[A,C] + [B, D] = [A, C, B, D]` when it could result in `[A, B, C, D]` to reduce search complexity later on. 

please dismiss if this is "premature optimisation".

Reply:
oh! in that case this may be more memory efficient and easier to read/reason, no?
```suggestion
				acc[key] = (record1?.[key] || []).concat(record2?.[key] || []);
```

Reply:
Thanks for the suggestion. Code fixed.